File: RewardedAdExample/ObjectiveCImplementation/RewardedAdExample/RewardedAdExample/ViewController.m
//
//  ViewController.m
//  RewardedAdExample
//
//  Created by Rishabh Tripathi on 19/11/24.
//

#import "ViewController.h"
#import "ApMobileSDK/ApMobileSDK.h"

@interface ViewController () <ApRewardedListener>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (IBAction)showAdBtn:(UIButton *)sender {
    [ApAppSDK showRewardedAdWithViewController:self apPlacementID:@"testPlacement" apRewardedListener:self];
}

- (void)onComplete {
    NSLog(@"Ad Complete");
}

@end

File: RewardedAdExample/SwiftImplementation/RewardedAdExample/ViewController.swift
//
//  ViewController.swift
//  RewardedAdExample
//
//  Created by Rishabh Tripathi on 04/11/24.
//

import UIKit
import ApMobileSDK

class ViewController: UIViewController, ApRewardedListener {
    func onComplete() {
        print("On Complete")
    }
    
    func onAdDismissed() {
        print("On Ad Dismissed")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    @IBAction func showAdTapped(_ sender: UIButton) {
        ApAppSDK.showRewardedAd(viewController: self, apPlacementID: "testPlacement", apRewardedListener: self)
    }
}

